### DISCUSSION TITLE
Random effect structure – using sdmTMB for a sparse river network.

### DISCUSSION THREAD

**Discussion by JulieCharbonneau** [sdmTMB/sdmTMB#379] [Q&A] [answered]
[sdm_angler_effort.docx](https://github.com/user-attachments/files/17726762/sdm_angler_effort.docx)

Hi there,
For part of my thesis work I am using an sdmTMB model to analyze steelhead angler effort (angler days/year/stream) over time across British Columbia. My goals are to:
1.	Assess whether angler effort has shifted over time and spatially across the province (using COG).
2.	Identify what factors drive shifts in effort (direction and magnitude of fixed effects).

The analysis includes 214 streams, however the data are not spatially explicit, and are only available at the stream level. To create the mesh, I selected one point for each stream 500 river meters from the mouth of the river.

I’m looking for guidance on selecting the random effect structure for this model, and understanding why the direction and magnitude of one of my fixed effects (population density) varies with different random effect structures. 

Initially, I had included a random effect of stream (214 levels), since the response (angler effort) is collected at the yearly level of a given stream. However, I was worried that since the mesh was constructed with those same 214 it would soak up too much of the variability and we would not be able to detect spatial shifts in effort (the spatial SD was 0). When running the model with a ranef of broader management area (5 levels: Van Island, Lower Mainland, Skeena, Cariboo, Thompson-Nicola) both the sanity checks and fit look better. However, the direction and magnitude of the effect of population density (i.e, whether angler effort is higher in more densely populated areas) changes drastically among models with different fixed effect structures. The model with stream as the random effect provides a result that is more aligned with both the hypothesis that there would be higher effort in more populated areas, and also the raw data, however the both the sanity checks and fit are poorer. I’ve attached a document with more details on the data and model outputs.

If anyone has insights into why this might be happening —or general advice on choosing the best random effect structure/model configuration in this context—I’d be very grateful! 😊

**Comment by ericward-noaa** [comment, marked as answer]
This is a good question. You have spatial fields included in all of these models, and those spatial fields (as random effects) are going to try to explain all of the covariates that might be related to the response. Adding on additional random effects that are spatially structured (such as stream or area) is generally not a good idea, because it can change the interpretation / effects of fixed effect variables. I think it's fine to include either, but if you did so I'd turn the spatial field off with `spatial="off"`

Looking at these models, you're very close with the 3rd model that has no random effects (your output pasted below). The ar1 phi gradient can be solved either by doing the extra optimization, adding a prior (see `?sdmTMBpriors`), or changing the model to a random walk. I'd start with this 3rd option because it looks like the AR1 phi parameter is 1.0 -- switching the model to a random walk (`spatiotemporal = "rw"`) just fixes this at 1 rather than estimating it.

As a larger check, it'd be worth also adding the model with no spatial fields, and an even simpler version with no spatial or spatiotemporal fields.

<img width="468" alt="image" src="https://github.com/user-attachments/assets/9b80d50e-024c-442a-b187-d7fc64ca25de">

**Reply by JulieCharbonneau** [reply]
Thank you so much for your help Eric, I really appreciate it! :)  I re-ran the model with "rw" instead of "ar1" and now it passes the sanity check.

Do you have any thoughts on why it could be that direction and magnitude of the one variable (population density) switches when there is no stream-specific random effect? The only problem with the model without the ranef, is that although it passes the sanity check, the estimate doesn't really make sense given the raw data (attached) or the hypothesis that you would expect higher angler effort around densely populated areas. I did try to smooth it, as you can see from the raw data it looks like there is some non-linearity in the response, but that also generates a large negative value. I find it especially strange, since the effect size of the other fixed effects does not change considerably among the different model configurations. 

[angler_effort_pop_dens.docx](https://github.com/user-attachments/files/17737756/angler_effort_pop_dens.docx)

**Reply by ericward-noaa** [reply]
Great to hear things are passing! 

I think most of your covariates are spatially structured, so I wonder if there's not some confounding that's not super straightforward (if you haven't looked at the correlations among variables, I'd definitely look at that). For population density, I would probably log transform it, if you haven't already -- the distribution looks to be pretty skewed. 

One of the other interesting features about this model is that the magnitude of estimated spatial variation is 50-100x higher than spatiotemporal variation. This suggests that the spatial field (representing variables that aren't changing in time) are explaining most of the variation in effort. It'd be interesting to plot this field -- if the spatial pattern is similar to population density (peaks near Vancouver for example) I think I'd try the model without the spatial field included, because the spatial field is fighting with your fixed effects to explain the same information.

**Comment by James-Thorson** [comment]
It might be interesting to compare this SPDE spatial model with an alternative using stream network distances ... feel free to look at the `sfnetworks` integration in `tinyVAST`, which would allow you to compare the Euclidean and stream-network distance metrics.
